code/datums/crew_objective.dm

```diff
@@ -456,56 +456,29 @@ ABSTRACT_TYPE(/datum/objective/crew/bartender)
 #define PIZZA_OBJ_COUNT 3
 ABSTRACT_TYPE(/datum/objective/crew/chef)
 /datum/objective/crew/chef
-	var/static/list/blacklist = list(
-		/obj/item/reagent_containers/food/snacks/burger/humanburger,
-		/obj/item/reagent_containers/food/snacks/donut/custom/robust,
-		/obj/item/reagent_containers/food/snacks/ingredient/meat/humanmeat,
-		/obj/item/reagent_containers/food/snacks/ingredient/meat/mysterymeat/nugget/flock,
-		/obj/item/reagent_containers/food/snacks/ingredient/pepperoni,
-		/obj/item/reagent_containers/food/snacks/meatball,
-		/obj/item/reagent_containers/food/snacks/mushroom,
-		/obj/item/reagent_containers/food/snacks/pickle/trash,
-		/obj/item/reagent_containers/food/snacks/pizza/xmas,
-		/obj/item/reagent_containers/food/snacks/plant/glowfruit/spawnable,
-		/obj/item/reagent_containers/food/snacks/soup/custom,
-		/obj/item/reagent_containers/food/snacks/condiment/syndisauce,
-		/obj/item/reagent_containers/food/snacks/donkpocket_w,
-		/obj/item/reagent_containers/food/snacks/surstromming,
-		/obj/item/reagent_containers/food/snacks/hotdog/syndicate,
-		/obj/item/reagent_containers/food/snacks/dippable/tortilla_chip_spawner,
-		/obj/item/reagent_containers/food/snacks/pancake/classic,
-		/obj/item/reagent_containers/food/snacks/wonton_spawner,
-		/obj/item/reagent_containers/food/snacks/agar_block,
-		/obj/item/reagent_containers/food/snacks/sushi_roll/custom,
-#ifndef UNDERWATER_MAP
-		/obj/item/reagent_containers/food/snacks/healgoo,
-		/obj/item/reagent_containers/food/snacks/greengoo,
-#endif
-		/obj/item/reagent_containers/food/snacks/snowball,
-		/obj/item/reagent_containers/food/snacks/burger/vr,
-		/obj/item/reagent_containers/food/snacks/slimjim,
-		/obj/item/reagent_containers/food/snacks/bite,
-		/obj/item/reagent_containers/food/snacks/pickle_holder,
-		/obj/item/reagent_containers/food/snacks/snack_cake,
-		/obj/item/reagent_containers/food/snacks/ice_cream/random,
-		/obj/item/reagent_containers/food/snacks/ice_cream/goodrandom
-	)
-	var/static/list/ingredients = concrete_typesof(/obj/item/reagent_containers/food/snacks) - blacklist - concrete_typesof(/obj/item/reagent_containers/food/snacks/ingredient/egg/critter)
 /datum/objective/crew/chef/cake
 	var/choices[CAKE_OBJ_COUNT]
 	var/completed = FALSE
 
 	set_up()
 		..()
 		var/list/names[CAKE_OBJ_COUNT]
-		for(var/i in 1 to CAKE_OBJ_COUNT)
-			choices[i] = pick(ingredients)
+		var/i = 0
+		var/current_rolls = 0
+		var/max_rolls = 30
+		while (i < CAKE_OBJ_COUNT)
+			i++
+			choices[i] = pick(allowed_favorite_ingredients)
 			var/choiceType = choices[i]
 			var/obj/item/reagent_containers/food/snacks/instance =  new choiceType
-			if(!instance.custom_food)
+			if(instance.custom_food)
+				names[i] = instance.name
+			else
 				i--
-				continue
-			names[i] = instance.name
+			current_rolls++
+			if (current_rolls > max_rolls)
+				logTheThing(LOG_DEBUG, "Failed to generate a foodlist objective for chef. Aborting.")
+				break
```

Comment:
I think this should be returning? since it'll continue generating ingredients even if it failed which we probably don't want.
```suggestion
				return
```